src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -208,8 +207,6 @@ public override void OnStopActivity(Activity activity, object payload)
                 // the one created by the instrumentation.
                 // And retrieve it here, and set it to Current.
             }
-
-            this.activitySource.Stop(activity);
```

Comment:
It looks like AspNetCore hosting keeps a reference to the Activity it started and that's the one it will stop:

https://github.com/dotnet/aspnetcore/blob/4a7ed303fff26e4070e02e385fb22df484f6b81f/src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs#L134-L139

So my question is: Does the Activity we create in this approach actually get stopped?

Reply:
@CodeBlanch , when you say "this approach", that means the old approach or this new one? For the new one, we are replacing the activity, but, if aspnetcore has a reference there, we might be creating a new one that would never be stopped.
@cijothomas to comment as well :)

Reply:
@eddynaka I meant the new approach sorry 😄

> For the new one, we are replacing the activity, but, if aspnetcore has a reference there, we might be creating a new one that would never be stopped.

Ya, exactly. That would be a problem!

Reply:
we are stopping our own activity. aspnet core stops their activity. check line 197 for our activity being stopped.

(This is a diff. issue in AspNet, as it relies on Activity.Current, so we need to restore it for AspNet, but for AspNetCore we dont. But agree - this is a bit flaky design, and will fall apart if aspnetcore changes their implementation. We'll need to revisit this in .NET6, when they are expected to do some changes.)

Reply:
@eddynaka Can you verify the one we're creating is stopped correctly? I see the `activity.Stop` on LN197 but we're stopping the same activity passed in OnStopActivity so it should already be stopped? :shrug:

Reply:
OnStopActivity gets Activity.Current, which should be the one we created, right?

Reply:
Ya you're right it works fine. Just took it for a spin to verify.